src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -532,5 +532,17 @@ public static object ConvertValuesToCollectionType<T>(Type modelType, IList<T> v
                 return null;
             }
         }
+
+        private static Type[] GetTypeArgumentsIfMatch([NotNull] Type closedType, Type matchingOpenType)
```

Comment:
The semantics are different in this method - we might be dealing with non-interface types (such as `KeyValuePair<>`).